tests/http/test_fedclient.py

```diff
@@ -268,6 +268,60 @@ def test_client_headers_no_body(self):
 
         self.assertIsInstance(f.value, TimeoutError)
 
+    def test_client_requires_trailing_slashes(self):
```

Comment:
nice test, thanks.

(might be nice to test that we *don't* retry on other sorts of errors, cf https://github.com/matrix-org/synapse/pull/4840/files#diff-eaf1ef90b23076afc24efff2afdfe772R232)